Core/Core/CourseSync/CourseSyncDownloader/Model/CourseSyncPeopleInteractor.swift

```diff
@@ -0,0 +1,148 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import CombineExt
+
+protocol CourseSyncPeopleInteractor: CourseSyncContentInteractor {}
+
+class CourseSyncPeopleInteractorLive: CourseSyncPeopleInteractor {
+
+    var associatedTabType: TabName { .people }
+
+    func getContent(courseId: String) -> AnyPublisher<Void, Error> {
+
+        let context: Context = .course(courseId)
+
+        return [
+            Self.fetchCourseColors(),
+            Self.fetchCourse(context: context),
+            Self.fetchSections(courseID: courseId),
+            Self.fetchUsers(context: context)
+                .flatMap { users in
+                    Self.fetchCurrentGradingPeriodId(courseID: courseId)
+                        .flatMap {
+                            Self.fetchUserData(context: context, users: users, currentGradingPeriodID: $0)
+                    }
+                }.eraseToAnyPublisher(),
+        ]
+            .zip()
+            .mapToVoid()
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchUserData(context: Context, users: [User], currentGradingPeriodID: String?) -> AnyPublisher<Void, Error> {
+        Just(users)
+            .map { $0.map { user in user.id} }
+            .flatMap { userIDs in
+                Publishers.Sequence(sequence: userIDs)
+                    .setFailureType(to: Error.self)
+                    .flatMap {
+                        Publishers.Zip3(Self.fetchSingleUser(context: context, userID: $0),
+                                        Self.fetchSubmissionsForStudent(context: context, userID: $0),
+                                        Self.fetchEnrollments(context: context, currentGradingPeriodID: currentGradingPeriodID, userID: $0) )
+                    }.collect().mapToVoid()
+            }.eraseToAnyPublisher()
+    }
+
+    private static func fetchCourseColors() -> AnyPublisher<Void, Error> {
+        ReactiveStore(useCase: GetCustomColors())
+            .getEntities()
+            .mapToVoid()
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchCourse(context: Context) -> AnyPublisher<Void, Error> {
+        ReactiveStore(useCase: GetCourse(courseID: context.id))
+            .getEntities()
+            .mapToVoid()
+            .eraseToAnyPublisher()
+    }
+
+    private static func fetchSubmissionsForStudent(context: Context, userID: String) -> AnyPublisher<Void, Error> {
+        guard userID == AppEnvironment.shared.currentSession?.userID else { return Just(()).setFailureType(to: Error.self).eraseToAnyPublisher() }
```

Comment:
I think a new line after/before `else` would make it more readable.

Reply:
done